src/AutomatedCar/Models/YamlNpcDocumentLoader.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using YamlDotNet.RepresentationModel;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace AutomatedCar.Models
+{
+    public class YamlNpcDocumentLoader : INpcDocumentLoader
+    {
+        public void LoadNpcDocument(Stream inputStream, INpcCreator npcCreator)
+        {
+            string inputStreamInString = "";
+            int TypeCount = 0;
+
+            using (StreamReader sr = new StreamReader(inputStream))
+            {
+                string line;
+                while ((line = sr.ReadLine()) != null)
+                {
+                    if (line.Contains("typename"))
+                    {
+                        TypeCount++;
+                    }
+                    inputStreamInString += line;
+                    inputStreamInString += "\n";
+                }
+            }
+
+            IList<YamlMappingNode> mapping = CreateMapping(inputStreamInString, TypeCount);
+            Dictionary<string, IEnumerable> npcsInDictionary = CreateDictionaryFromMapping(mapping);
+
+            CreateNpcs(npcsInDictionary, npcCreator);
+        }
+
+        public void CreateNpcs(Dictionary<string, IEnumerable> npcsInDictionary, INpcCreator npcCreator)
+        {
+            foreach (var npcObj in npcsInDictionary)
+            {
+                IList<PathNode> path = new List<PathNode>();
+                foreach (PathNode node in npcObj.Value)
+                {
+                    path.Add(node);
+                }
+                npcCreator.Create(npcObj.Key, path);
+            }
+        }
+
+        public Dictionary<string, IEnumerable> CreateDictionaryFromMapping(IList<YamlMappingNode> mapping)
+        {
+            Dictionary<string, IEnumerable> npcs = new Dictionary<string, IEnumerable>();
+
+            foreach (var npcObj in mapping)
+            {
+                var pathInYaml = (YamlSequenceNode)npcObj.Children[new YamlScalarNode("path")];
+                IList<PathNode> path = new List<PathNode>();
+                foreach (var node in pathInYaml)
+                {
+                    PathNode n = new PathNode();
+
+                    n.Rotation = float.Parse(node[new YamlScalarNode("rot").Value].ToString(), CultureInfo.InvariantCulture.NumberFormat);
+                    n.Speed = float.Parse(node[new YamlScalarNode("speed").Value].ToString(), CultureInfo.InvariantCulture.NumberFormat);
+                    
+                    string s = (node[new YamlScalarNode("pos").Value]).ToString();
+                    float x = float.Parse(s.Split("x, ")[1].Split(' ')[0]);
+                    float y = float.Parse(s.Split("y, ")[1].Split(' ')[0]);
+                    n.Position = new Vector2(x, y);
```

Comment:
A path node 'pos' tulajdonsága nem egy string amit kézzel kéne feldolgozni, hanem egy sima mapping.
```suggestion
                    var nodePos = node["pos"] as YamlMappingNode;
                    var nodeX = nodePos["x"] as YamlScalarNode;
                    var nodeY = nodePos["y"] as YamlScalarNode;
                    var x = float.Parse(nodeX.Value, CultureInfo.InvariantCulture.NumberFormat);
                    var y = float.Parse(nodeY.Value, CultureInfo.InvariantCulture.NumberFormat);
                    n.Position = new Vector2(x, y);
```